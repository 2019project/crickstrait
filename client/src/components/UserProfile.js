import React, { Component } from "react";
import "../styles/UserProfile.css";
import UserNavigation from "./Common/UserNavigation";
import { connect } from "react-redux";
import { searchContent } from "../actions/Navbar";
import { buyStocks } from "../actions/Users";
import jwt_decode from "jwt-decode";
import { Link } from "react-router-dom";

export class UserProfile extends Component {
  state = {
    // input text in the search box
    searchInput: "",
    searchInputChanged: false,
    qty: "",
    total: 0
  };

  OnChange = event => {
    {
      this.state.searchInputChanged
        ? console.log(this.state.searchInputChanged)
        : this.setState({
            searchInputChanged: true
          });
    }
    this.setState({ [event.target.name]: event.target.value });

    let searchString = {
      searchInput: this.state.searchInput
    };
    this.props.searchContent(searchString);
  };

  onSearch = e => {
    e.preventDefault();

    let searchString = {
      searchInput: this.state.searchInput
    };

    this.props.searchContent(searchString);
  };

  render() {
    if (localStorage.getItem("token")) {
      var decode = jwt_decode(localStorage.getItem("token"));
    }
    console.log(this.props.results);
    return (
      <div>
        <div
          id={
            localStorage.getItem("token")
              ? "analysis_main_container"
              : "analysis_main_container_blur"
          }
        >
          <div id="userProfileContainer">
            <UserNavigation wallet={this.props.users} />

            <div id="userSearch">
              <h1>Welcome to HeadStocks</h1>
              <input
                type="text"
                placeholder="Search for Stocks you want to buy (E.g . AAPL)"
                name="searchInput"
                id="userProfileSearch"
                value={this.state.searchInput}
                onChange={this.OnChange}
              />
              <button type="submit" id="userProfileSearchButton">
                <i class="fa fa-search"></i>
              </button>

              {this.state.searchInput == "" ? null : (
                <div className="search-result">
                  {this.props.results.length != 0 ? (
                    this.props.results.map((stocks, index) => (
                      <div id="search_map">
                        <div
                          id={"tickerName" + index}
                          style={{
                            fontSize: "18px",
                            paddingTop: "0px ",
                            display: "flex",
                            justifyContent: "center"
                          }}
                        >
                          <b>{stocks.ticker_name}</b>
                        </div>
                        <div>
                          <p id={"tickerPrice" + index}>
                            <b>${stocks.price}</b>
                          </p>
                          <p style={{ color: "#707070" }}>Current Price</p>
                        </div>
                        <div>
                          <div class="buyStocksBox">
                            <a class="buy" href="#buyStockspopup1">
                              Buy
                            </a>
                          </div>

                          <div id="buyStockspopup1" class="buyStocksoverlay">
                            <div class="buyStockspopup">
                              <h2>{stocks.ticker_name}</h2>
                              <a class="buyStocksclose" href="#">
                                &times;
                              </a>
                              <div class="buyStockscontent">
                                <p>Current Price : ${stocks.price}</p>
                                <p>
                                  Quantity :{" "}
                                  <input
                                    type="text"
                                    // class="quantity"
                                    id="buyingQuantity"
                                    name="qty"
                                    value={this.state.qty}
                                    onChange={e =>
                                      this.setState({
                                        [e.target.name]: e.target.value,
                                        total: (e.target.value *= Number(
                                          stocks.price
                                        ))
                                      })
                                    }
                                  />
                                </p>
                                <p>
                                  Total Price : $
                                  {Number(this.state.total).toFixed(2)}
                                </p>
                                <a
                                  id="buySpecificStock"
                                  href="#"
                                  onClick={() => {
                                    {
                                      let user = {
                                        email: decode.email,
                                        ticker_name: stocks.ticker_name,
                                        current_price: stocks.price,
                                        qty: this.state.qty,
                                        price: this.state.total
                                      };
                                      this.props.buyStocks(user);
                                    }
                                  }}
                                >
                                  Buy
                                </a>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    ))
                  ) : (
                    <p style={{ padding: "10px 10px" }}>No Company Found</p>
                  )}
                </div>
              )}
            </div>
          </div>
        </div>
        <div id={localStorage.getItem("token") ? "bg-no-text" : "bg-text"}>
          <p>Please login to enable this feature</p>
          <Link to="/login">
            <button id="button_for_blur">Login</button>
          </Link>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  results: state.navbarReducer.results,
  isLoading: state.LoadingReducer.isLoading,
  users: state.usersReducer.users
});

export default connect(mapStateToProps, { searchContent, buyStocks })(
  UserProfile
);
