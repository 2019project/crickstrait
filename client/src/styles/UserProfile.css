@import url(https://fonts.googleapis.com/css?family=Karla);
* {
  font-family: Karla;
}

#userProfileContainer {
  width: 100%;
  display: grid;
  grid-template-rows: 150px auto;
  text-align: center;
}

#userSearch {
  background: url("../components/Common/stocks_wallpaper.jpg") no-repeat;
  background-size: cover;
  margin-top: 60px;
  height: 92vh;
}

#userSearch h1 {
  text-align: center;
  font-weight: 700;
  font-family: Karla;
  margin-top: 100px;
  color: #2c3e50;
  text-shadow: 4px 4px 6px #bebebe;
}

#userProfileSearch {
  width: 67%;
  box-shadow: 4px 4px 6px #707070;
  color: #707070;
  border-top-left-radius: 8px;
  border-bottom-left-radius: 8px;
  padding: 10px;
  margin-top: 8px;
  font-size: 20px;
  font-family: Karla;
  outline: none;
  border: none;
}

/* Serach result */
.search-result {
  border: 1px solid #c2c2c2;
  height: 150px;
  overflow-y: scroll;
  width: 78%;
  margin: 5px 0px 0px 27px;
  border-radius: 2px;
  background-color: white;
}

#userProfileSearchButton {
  box-shadow: 4px 4px 6px #707070;
  padding: 10px 10px;
  margin-top: 8px;
  margin-right: 16px;
  background: #fff;
  border-top-right-radius: 8px;
  border-bottom-right-radius: 8px;
  font-size: 20px;
  border: none;
  cursor: pointer;
  color: #707070;
}

#buyButton {
  padding: 12px 20px;
  color: #fff;
  border-radius: 25px;
  cursor: pointer;
  transition: all 0.3s ease-out;
  background-color: #39abf7;
  color: #fff;
  border: none;
  text-decoration: none;
  outline: none;
  margin-top: 100px;
}

.buyStocksBox {
  margin: 0 auto;
  border: 2px solid #fff;
  text-align: center;
}

.buy {
  font-size: 1em;
  border-radius: 20px;
  padding: 10px 20px;
  color: #fff;
  background-color: #2c3e50;
  text-decoration: none;
  cursor: pointer;
  transition: all 0.3s ease-out;
}
.buy:hover {
  background: #39abf7;
}

.buyStocksoverlay {
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  background: rgba(0, 0, 0, 0.7);
  transition: opacity 500ms;
  visibility: hidden;
  opacity: 0;
}
.buyStocksoverlay:target {
  visibility: visible;
  opacity: 1;
}

.buyStockspopup {
  margin: 150px auto;
  padding: 8px;
  background: #2c3e50;
  border-radius: 5px;
  width: 95%;
  position: relative;
  transition: all 5s ease-in-out;
}

.buyStockspopup h2 {
  margin-top: 0;
  color: #fff;
  font-family: Tahoma, Arial, sans-serif;
  font-weight: 500;
  font-size: 40px;
}
.buyStockspopup .buyStocksclose {
  position: absolute;
  top: 20px;
  right: 30px;
  transition: all 200ms;
  font-size: 30px;

  font-weight: bold;
  text-decoration: none;
  color: #fff;
}
.buyStockspopup .buyStocksclose:hover {
  color: #39abf7;
  cursor: pointer;
}
.buyStockspopup .buyStockscontent {
  max-height: 30%;
  overflow: auto;
  color: #fff;
}

.buyStockspopup .buyStockscontent p {
  font-size: 18px;
  padding: 10px;
}

#search_map {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  height: 50px;
}

#search_map > div {
  margin: auto;
  font-size: 12px;
  color: #2c3e50;
}

#buyingQuantity {
  width: 15%;
  height: 20px;
  line-height: 1.65;
  background: transparent;
  padding: 10px;
  color: #fff;
  border: 2px solid #39abf7;
  border-radius: 4px;
}

#buyingQuantity::placeholder {
  color: #fff;
}

#buySpecificStock {
  border: 2px solid #39abf7;
  padding: 5px 20px;
  color: #fff;
  outline: none;
  font-family: Karla;
  cursor: pointer;
  margin-top: 20px;
  border-radius: 4px;
  background: transparent;
}

#buySpecificStock:hover {
  right: 0;
  background-color: #39abf7;
  box-shadow: 0 12px 16px 0 rgba(231, 226, 226, 0.24),
    0 17px 50px 0 rgba(231, 226, 226, 0.24);
}

@media screen and (min-width: 430px) {
  #userProfileContainer {
    width: 100%;
    grid-template-columns: 250px 1fr;
    font-family: Karla;
    margin: auto;
  }

  #userSearch {
    margin-top: 0px;
  }
  .buyStockspopup {
    margin: 250px auto;
    padding: 20px;
    background: #2c3e50;
    border-radius: 5px;
    width: 30%;
    position: relative;
    transition: all 5s ease-in-out;
  }

  .search-result {
    border: 1px solid #c2c2c2;
    height: auto;
    max-height: 400px;
    overflow-y: scroll;
    width: 70%;
    margin: 5px 0px 0px 182px;
    border-radius: 2px;
    background-color: white;
  }

  #search_map > div {
    margin: auto;
    font-size: 15px;
  }

  #search_map {
    width: 100%;
    height: 60px;
    box-shadow: 0px 1px 2px #cacaca;
  }

  #search_map:hover {
    box-shadow: 0px 2px 8px #2c3e50;
  }
}
