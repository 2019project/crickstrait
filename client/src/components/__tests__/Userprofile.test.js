import React from "react";
import { shallow, mount } from "enzyme";
import { UserProfile } from "../UserProfile";

const userProfile = jest.fn();
const searchContent = jest.fn();
const results = [
  {
    industry: "Retail - Apparel & Specialty",
    price: 156.59,
    ticker_id: 8,
    ticker_name: "AAP"
  },
  {
    industry: "Retail - Apparel & Specialty",
    price: 156.59,
    ticker_id: 8,
    ticker_name: "AAP"
  }
];
const wrapper = shallow(
  <UserProfile
    userProfile={userProfile}
    searchContent={searchContent}
    results={results}
  />
);

describe("Testing User Profile Component", () => {
  it("should mount the component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("shoudl check for input value", () => {
    const e = {
      target: {
        name: "searchInput",
        value: "aapl"
      }
    };
    expect(wrapper.state().searchInput).toBe("");
    wrapper.instance().OnChange(e);
    expect(wrapper.state().results).toBe(undefined);
    expect(searchContent).toBeCalledTimes(1);
  });

  it("checks for on Buy Stocks to be called", () => {
    const e = { preventDefault: () => {} };
    jest.spyOn(e, "preventDefault");
    wrapper.find("#buySpecificStock").simulate("click", e);
  });

  it("should map tickers from search result", () => {
    expect(wrapper.find("#tickerName0").props().children).toBe(
      results.ticker_name
    );
  });
});
