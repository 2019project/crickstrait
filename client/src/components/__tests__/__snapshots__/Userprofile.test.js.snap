// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing User Profile Component should mount the component 1`] = `
<div>
  <div
    id="userProfileContainer"
  >
    <UserProfile />
    <div
      id="userSearch"
    >
      <h1>
        Welcome to HeadStocks
      </h1>
      <input
        id="userProfileSearch"
        name="searchInput"
        onChange={[Function]}
        placeholder="Search for Stocks you want to buy (E.g . AAPL)"
        type="text"
        value=""
      />
      <button
        id="userProfileSearchButton"
        type="submit"
      >
        <i
          class="fa fa-search"
        />
      </button>
    </div>
  </div>
</div>
`;
